tests/test_trait_pipelines.py

```diff
@@ -593,78 +595,373 @@ def test_OlderMonocot_pipeline(rice_main_10do_h5, rice_main_10do_slp):
-        "crown_count": "int64",
-    }
+    batch_traits_computed = pipeline.compute_batch_traits(
+        [series_0K9E8BI, series_YR39SJX]
+    )
+    batch_traits_csv = pd.read_csv(rice_3do_batch_traits_csv)
 
-    expected_all_traits_dtypes = {
-        "crown_count_min": "int64",
-        "crown_count_max": "int64",
+    # Shape check.
+    assert traits_computed_0K9E8BI.shape == (72, 104)
+    assert traits_computed_YR39SJX.shape == (72, 104)
+    assert batch_traits_computed.shape == (2, 919)
+
+    expected_dtypes = (int, float, np.integer, np.floating)
+
+    monocot_dfs = {}
+
+    for series in all_series:
+        traits_records = []
+        for frame_idx in range(72):
+            # Calculate traits.
+            trait_dict = {
+                "plant_name": series.series_name,
+                "frame_idx": frame_idx,
+                "primary_pts": series.get_primary_points(frame_idx),
+                "crown_pts": series.get_crown_points(frame_idx),
+            }
+
+            trait_dict["primary_max_length_pts"] = get_max_length_pts(
+                trait_dict["primary_pts"]
+            )
+            trait_dict["pts_all_array"] = get_all_pts_array(trait_dict["crown_pts"])
+            trait_dict["crown_count"] = get_count(trait_dict["crown_pts"])
+            trait_dict["crown_proximal_node_inds"] = get_node_ind(
+                trait_dict["crown_pts"], proximal=True
+            )
+            trait_dict["crown_distal_node_inds"] = get_node_ind(
+                trait_dict["crown_pts"], proximal=False
+            )
+            trait_dict["crown_lengths"] = get_root_lengths(trait_dict["crown_pts"])
+            trait_dict["crown_base_pts"] = get_bases(trait_dict["crown_pts"])
+            trait_dict["crown_tip_pts"] = get_tips(trait_dict["crown_pts"])
+            trait_dict["scanline_intersection_counts"] = count_scanline_intersections(
+                trait_dict["crown_pts"],
+                height=pipeline.img_height,
+                n_line=pipeline.n_scanlines,
+            )
+            trait_dict["crown_angles_distal"] = get_root_angle(
+                trait_dict["crown_pts"],
+                trait_dict["crown_distal_node_inds"],
+                proximal=False,
+                base_ind=0,
+            )
+            trait_dict["crown_angles_proximal"] = get_root_angle(
+                trait_dict["crown_pts"],
+                trait_dict["crown_proximal_node_inds"],
+                proximal=True,
+                base_ind=0,
+            )
+            trait_dict["bounding_box"] = get_bbox(trait_dict["crown_pts"])
+            trait_dict["network_length_lower"] = get_network_distribution(
+                trait_dict["crown_pts"],
+                trait_dict["bounding_box"],
+                fraction=pipeline.network_fraction,
+            )
+            trait_dict["ellipse"] = fit_ellipse(trait_dict["crown_pts"])
+            trait_dict["convex_hull"] = get_convhull(trait_dict["crown_pts"])
```

Comment:
why does this work? I thought we had to use `pts_all_array` for convex hull, ellipse, bounding box etc.